backend/back_app/users/serializers.py

```
@@ -43,3 +47,49 @@ def create(self, validated_data):
         validated_data.pop("password2")
         user = User.objects.create_user(**validated_data)
         return user
+
+
+class PasswordResetRequestSerializer(serializers.Serializer):
+    email = serializers.EmailField()
+
+    def validate_email(self, value):
+        if not User.objects.filter(email=value).exists():
+            raise serializers.ValidationError("User with this email does not exist")
+        return value
+
+    def send_reset_email(self):
+        email = self.validated_data["email"]
+        user = User.objects.get(email=email)
+        token = default_token_generator.make_token(user)
+        uid = urlsafe_base64_encode(force_bytes(user.pk))
+        reset_url = (
+            f"{settings.URL_NGROK_HOST}/users/reset-password-confirm/{uid}/{token}/"
+        )
+        send_mail(
+            subject="Password Reset Request",
+            message=f"Use the link to reset your password: {reset_url}",
+            from_email=None,
+            recipient_list=[email],
+        )
+        return reset_url
+
+
+class PasswordResetConfirmSerializer(serializers.Serializer):
+    new_password = serializers.CharField(write_only=True)
+
+    def validate(self, attrs):
+        uidb64 = self.context.get("uidb64")
+        token = self.context.get("token")
+        try:
+            uid = force_str(urlsafe_base64_decode(uidb64))
+            self.user = User.objects.get(pk=uid)
+        except (TypeError, ValueError, User.DoesNotExist):
```

@Sn1kls Split these errors:

User.DoesNotExist should return a 404 error when something is not found.
For other errors related to user-entered data, use a 400 error as a generic client error status.